test/inductor/test_mkldnn_pattern_matcher.py

```diff
@@ -110,7 +127,17 @@ def cal_conv_generated_kernel_number(mod, input, dtype):
     ):
         input_kernel = 1
     if output.is_contiguous(memory_format=torch.contiguous_format):
-        output_kernel = 1
+        # For xpu dynamic conv, we don't do layout optimization in fw_compile_freezing:
+        # https://github.com/pytorch/pytorch/blob/99a03211cb58cf3e5e3dea0c543795af43c41da8/torch/_inductor/graph.py#L528C1-L555
+        # Thus we don't need a new kernel to enforce output_layout.
```

Comment:
Just wondering why it is cpu specific? It seems to be mkldnn library specific?

Reply:
For cpu backend and mkldnn enabled, we always use channels_last for better performance. But for XPU the performance benefit have not been validated we'll try that.